.github/workflows/e2e-crc-okd-tests.yml

```diff
@@ -0,0 +1,109 @@
+  cancel-in-progress: true
+
+env:
+  IT_MODULE: kubernetes-itests
+  MAVEN_ARGS: -B -C -V -ntp -Dhttp.keepAlive=false -e
+  SHELL: /bin/bash
+
+jobs:
+  openshift-kubernetes-distribution:
+    name: CRC ${{ matrix.crc }} / OKD ${{ matrix.okd }}
+    runs-on: ubuntu-24.04
+    if: github.repository == 'fabric8io/kubernetes-client'
+    strategy:
+      fail-fast: false
+      matrix:
+        # There is some problem with latest version of crc configured with okd preset. I
+        # wasn't able to run tests successfully on latest version of crc. See linked issue:
+        # https://github.com/crc-org/crc/issues/4382
+        # TODO: Update CRC version to latest when aforementioned issue gets resolved
+        # https://github.com/fabric8io/kubernetes-client/issues/6415
+        okd: [ v4.14.0 ]
+        crc: [ 2.32.0 ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Java 11
+        uses: actions/setup-java@v4
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+      - name: Install required virtualization software
+        run: |
+          sudo apt-get update
+          sudo apt install qemu-kvm libvirt-daemon libvirt-daemon-system
+          sudo usermod -a -G libvirt $USER
+          UBUNTU_VERSION=`cat /etc/os-release | grep VERSION_ID | awk -F= '{print $2}' | tr -d '"'`
+          echo "Ubuntu version is $UBUNTU_VERSION"
+          if [ $UBUNTU_VERSION == "22.04" ]; then
+            echo "IT's 22.04, virtiofsd would be already installed"
+          elif [ $UBUNTU_VERSION == "24.04" ]; then
+            echo "IT's 24.04, installing virtiofsd"
+            sudo apt install virtiofsd
+          else
+            echo "None found, Actual $UBUNTU_VERSION"
+          fi
+      - name: Remove unwanted stuff to free up disk image
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /usr/local/lib/android
+          sudo rm -rf /opt/ghc
+          sudo rm -rf "/usr/local/share/boost"
+          sudo rm -rf /opt/hostedtoolcache/CodeQL
+
+          sudo docker image prune --all --force
+
+          sudo swapoff -a
+          sudo rm -f /mnt/swapfile
+          sudo adduser $USER libvirt
+          sudo adduser $USER kvm
+          sudo usermod -a -G libvirt $USER
+      - name: Download CRC
+        run: |
+          wget https://developers.redhat.com/content-gateway/file/pub/openshift-v4/clients/crc/${{ matrix.crc }}/crc-linux-amd64.tar.xz
+          tar -xJf crc-linux-amd64.tar.xz
+          sudo cp crc-linux-${{ matrix.crc }}-amd64/crc /usr/local/bin/crc
+      - name: Set the crc config
+        run: |
+          crc config set preset okd
+          crc config set network-mode user
+      - name: Setup the crc
+        run: |
+          sudo -su $USER crc setup
+      - name: Start the crc
+        run: sudo -su $USER crc start
+      - name: Install Kubernetes Client
+        run: ./mvnw ${MAVEN_ARGS} clean install -DskipTests -Djacoco.skip=true
```

Comment:
I`make quickly` should be used here instead.

Especially if we intend to run this for each PR